src/view/writer.js

```diff
@@ -1108,23 +1197,30 @@ function rangeSpansOnAllChildren( range ) {
 
 // Checks if provided nodes are valid to insert. Checks if each node is an instance of
 // {@link module:engine/view/text~Text Text} or {@link module:engine/view/attributeelement~AttributeElement AttributeElement},
-// {@link module:engine/view/containerelement~ContainerElement ContainerElement} or
-// {@link module:engine/view/emptyelement~EmptyElement EmptyElement}.
+// {@link module:engine/view/containerelement~ContainerElement ContainerElement},
+// {@link module:engine/view/emptyelement~EmptyElement EmptyElement} or
+// {@link module:engine/view/uielement~UIElement UIElement}.
 //
 // Throws {@link module:utils/ckeditorerror~CKEditorError CKEditorError} `view-writer-insert-invalid-node` when nodes to insert
 // contains instances that are not {@link module:engine/view/text~Text Texts},
 // {@link module:engine/view/emptyelement~EmptyElement EmptyElements},
+// {@link module:engine/view/uielement~UIElement UIElements},
 // {@link module:engine/view/attributeelement~AttributeElement AttributeElements} or
 // {@link module:engine/view/containerelement~ContainerElement ContainerElements}.
 //
 // @param Iterable.<module:engine/view/text~Text|module:engine/view/attributeelement~AttributeElement
 // |module:engine/view/containerelement~ContainerElement> nodes
 function validateNodesToInsert( nodes ) {
 	for ( let node of nodes ) {
-		if ( !( node instanceof Text || node instanceof AttributeElement || node instanceof ContainerElement || node instanceof EmptyElement ) ) {
+		if (
+			!( node instanceof Text || node instanceof AttributeElement || node instanceof ContainerElement ||
```

Comment:
Here we can find some details about this condition: https://github.com/ckeditor/ckeditor5-engine/issues/377